### PR TITLE
chore: upgrade unified bridge SDK

### PR DESCRIPTION
**Pull request by meeh0w**
## Description
* [CP-9112](https://ava-labs.atlassian.net/browse/CP-9112)
* [CP-8047](https://ava-labs.atlassian.net/browse/CP-8047)

## TODOs
- [x] Release `@avalabs/bridge-unified` to a stable channel

## Changes
* Upgrades `@avalabs/bridge-unified` dependency
    * The new version includes some breaking changes. This PR updates our code to make it compatible.
* Refactors how we "recognise" CCTP (and the upcoming ICTT) bridge transactions - previously we would look at UnifiedBridge config to fetch the addresses, with this PR the updated SDK exposes the `analyzeTx()` method that we can use.
* Implements a schema migration for any pending bridge transfers (the `BridgeTransfer` object changed its shape, some properties changed names and/or places)

## Testing
* Test bridging. It should work exactly the same as it used to.
* Run sanity checks on other parts of the extension
* **Bonus points** for simulating the extension update (initiate bridging from `main`, switch to this branch and verify that your transaction is still tracked properly).

## Screenshots:
_Starts bridging via CCTP on `main`, simulates the extension update, then starts another bridge with the "updated" extension:_

https://github.com/user-attachments/assets/7768852c-398d-4fab-a876-ed72406c3ef2

## Checklist for the author
- [x] I've covered new/modified business logic with Jest test cases.
- [x] I've tested the changes myself before sending it to code review and QA.

### CODE REVIEW

File: src/contexts/UnifiedBridgeProvider.tsx [hunk 1]
```diff
@@ -476,13 +517,23 @@ export function UnifiedBridgeProvider({
...
+
   return (
     <UnifiedBridgeContext.Provider
       value={{
         estimateTransferGas,
         getErrorMessage,
         state,
-        getAssetAddressOnTargetChain,
+        isBridgeTx,
+        getAssetIdentifierOnTargetChain,
```
Comment (gergelylovas) [inline src/contexts/UnifiedBridgeProvider.tsx:536]: should we add tests for these?

**Review by gergelylovas — DISMISSED** [review]

File: src/contexts/UnifiedBridgeProvider.tsx (hunk 1 above)
Comment (meeh0w) [reply src/contexts/UnifiedBridgeProvider.tsx]: The `getAssetIdentifierOnTargetChain` will be removed as soon as ICTT is finalized, so I wouldn't bother personally.

`isBridgeTx` is now `analyzeTx` - I'll add tests in the upcoming PR (here: https://github.com/ava-labs/core-extension/pull/57)

File: src/components/common/approval/TransactionDetailItem.tsx [hunk 2]
```diff
@@ -12,6 +12,7 @@ import {
   type LinkItem,
   type TextItem,
   DetailItemType,
+  FundsRecipientItem,
```
Comment (gergelylovas) [inline src/components/common/approval/TransactionDetailItem.tsx:15]: Seems like you have some import errors here.

**Review by gergelylovas — CHANGES_REQUESTED** [review]

File: src/background/services/history/utils/isTxHistoryItem.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { TxHistoryItem } from '../models';
 
 export function isTxHistoryItem(tx: TxHistoryItem): tx is TxHistoryItem {
-  if ('isBridge' in tx && tx.vmType !== 'AVM' && tx.vmType !== 'PVM') {
+  if ('bridgeAnalysis' in tx && tx.vmType !== 'AVM' && tx.vmType !== 'PVM') {
```
[the remark this answers is not in the document]
Comment (meeh0w) [reply src/background/services/history/utils/isTxHistoryItem.ts]: I renamed it and rewrote the output signature to match the implementation in e93ba1c69.

I'm not 100% sure if its usage is always justified/correct, but chose to leave it as it is to limit the scope of this PR.

File: src/components/common/approval/TransactionDetailItem.tsx (hunk 2 above)
Comment (meeh0w) [reply src/components/common/approval/TransactionDetailItem.tsx:15]: Yeah, I pinned the wrong version for the `VM-` packages. Corrected in ce3b826dc

File: src/background/vmModules/ApprovalController.ts [hunk 4]
```diff
@@ -173,6 +173,7 @@ export class ApprovalController implements IApprovalController {
 
     switch (signingData.type) {
       case RpcMethod.BITCOIN_SEND_TRANSACTION:
+      case RpcMethod.BITCOIN_SIGN_TRANSACTION:
```
[the remark this answers is not in the document]
Comment (meeh0w) [reply src/background/vmModules/ApprovalController.ts:176]: Same as the other comment, pinned wrong version for `VM-` packages (fixed by ce3b826dc)

File: src/pages/Bridge/utils/blockchainConversion.test.ts [hunk 5]
```diff
@@ -1,18 +1,19 @@
...
+import { decorateWithCaipId } from '@src/utils/caipConversion';
 
 jest.mock('i18next', () => ({
   t: jest.fn(),
 }));
 
 describe('src/pages/Bridge/utils/blockchainConversion.ts', () => {
   const btcSymbol = 'BTC';
-  const mockBTCNetwork: Network = {
+  const mockBTCNetwork = decorateWithCaipId({
```
[the remark this answers is not in the document]
Comment (meeh0w) [reply src/pages/Bridge/utils/blockchainConversion.test.ts:16]: It just felt better to not leak the `BitcoinCaipId` out of the `caipConversion.ts` util -- but I'm happy to change if it's a strong opinion of yours :)

File: src/background/services/unifiedBridge/UnifiedBridgeService.ts [hunk 6]
```diff
@@ -104,51 +110,82 @@ export class UnifiedBridgeService implements OnStorageReady {
...
+          this.#getDisabledBridges()
+        ),
+      });
+      this.#failedInitAttempts = 0;
       this.#trackPendingTransfers();
-    });
+    } catch (err: any) {
+      // If it failed, it's most likely a network issue.
+      // Wait a bit and try again.
+      this.#failedInitAttempts += 1;
```
[the remark this answers is not in the document]
Comment (meeh0w) [reply src/background/services/unifiedBridge/UnifiedBridgeService.ts:143]: Added in da600aea9

File: src/contexts/UnifiedBridgeProvider.tsx [hunk 7]
```diff
@@ -124,25 +131,62 @@ export function UnifiedBridgeProvider({
...
+        ? [
+            BridgeType.CCTP,
+            BridgeType.ICTT_ERC20_ERC20,
+            BridgeType.AVALANCHE_EVM,
+          ]
+        : [BridgeType.AVALANCHE_EVM, BridgeType.ICTT_ERC20_ERC20],
     [isCCTPDisabled]
   );
 
+  const environment = useMemo(() => {
```
[the remark this answers is not in the document]
Comment (meeh0w) [reply src/contexts/UnifiedBridgeProvider.tsx:146]: Added in 5bf2855cf

**Review by gergelylovas — DISMISSED** [review]
👏

File: package.json [hunk 8]
```diff
@@ -23,10 +23,10 @@
     "sentry": "node sentryscript.js"
   },
   "dependencies": {
-    "@avalabs/avalanche-module": "0.11.2",
+    "@avalabs/avalanche-module": "^0.11.2",
```
Comment (gergelylovas) [inline package.json:26]: can you please pin the dependencies?

**Review by gergelylovas — CHANGES_REQUESTED** [review]

File: package.json (hunk 8 above)
Comment (meeh0w) [reply package.json]: 😱 done!

**Review by bferenc — APPROVED** [review]

**Review by gergelylovas — APPROVED** [review]

**Pull request merged**